pkg/disttask/framework/dispatcher/dispatcher.go

```diff
@@ -570,29 +566,29 @@ func (d *BaseDispatcher) onErrHandlingStage(receiveErrs []error) error {
 		subTasks = make([]*proto.Subtask, 0, len(instanceIDs))
 		for _, id := range instanceIDs {
 			// reverting subtasks belong to the same step as current active step.
-			subTasks = append(subTasks, proto.NewSubtask(d.Task.Step, d.Task.ID, d.Task.Type, id, d.Task.Concurrency, []byte("{}")))
+			subTasks = append(subTasks, proto.NewSubtask(
+				d.Task.Step, d.Task.ID, d.Task.Type, id,
+				d.Task.Concurrency, proto.EmptyMeta, 0))
 		}
 	}
 	return d.updateTask(proto.TaskStateReverting, subTasks, RetrySQLTimes)
 }
 
-func (*BaseDispatcher) nextStepSubtaskDispatched(*proto.Task) bool {
-	// TODO: will implement it when we we support dispatch subtask by batch.
-	// since subtask meta might be too large to save in one transaction.
-	return true
-}
-
 func (d *BaseDispatcher) onNextStage() (err error) {
-	/// dynamic dispatch subtasks.
-	failpoint.Inject("mockDynamicDispatchErr", func() {
-		failpoint.Return(errors.New("mockDynamicDispatchErr"))
-	})
-
 	nextStep := d.GetNextStep(d.Task)
 	logutil.Logger(d.logCtx).Info("onNextStage",
 		zap.Int64("current-step", int64(d.Task.Step)),
 		zap.Int64("next-step", int64(nextStep)))
 
+	if nextStep == proto.StepDone {
+		d.Task.Step = nextStep
+		d.Task.StateUpdateTime = time.Now().UTC()
+		if err = d.OnDone(d.ctx, d, d.Task); err != nil {
+			return errors.Trace(err)
+		}
+		return d.taskMgr.SucceedTask(d.ctx, d.Task.ID)
+	}
+
 	// 1. Adjust the task's concurrency.
```

Comment:
```suggestion
	// Adjust the task's concurrency.
```
Since we have no multi steps, remove seq.No